test/config-luatest/jsonschema_config_test.lua

```diff
@@ -0,0 +1,103 @@
+local cluster_config = require("internal.config.cluster_config")
+local config = require('config')
+local t = require('luatest')
+
+local g = t.group()
+
+local function remove_descriptions(tbl)
+    if type(tbl) ~= 'table' then
+        return
+    end
+
+    for key, value in pairs(tbl) do
+        if key == 'description' then
+            tbl[key] = nil
+        elseif type(value) == 'table' then
+            remove_descriptions(value)
+        end
+    end
+end
+
+g.test_cluster_config_schema_description_completeness = function()
+    local function check_schema_description(schema, ctx)
+        local field_path = table.concat(ctx.path, '.')
+        t.assert(schema.description ~= nil or field_path == '',
+                 string.format("%q is missing description", field_path))
+        if schema.type == 'record' then
+            for field_name, field_def in pairs(schema.fields) do
+                table.insert(ctx.path, field_name)
+                check_schema_description(field_def, ctx)
+                table.remove(ctx.path)
+            end
+        elseif schema.type == 'map' then
+            table.insert(ctx.path, '*')
+            check_schema_description(schema.value, ctx)
+            table.remove(ctx.path)
+        elseif schema.type == 'array' then
+            table.insert(ctx.path, '*')
+            check_schema_description(schema.items, ctx)
+            table.remove(ctx.path)
+        end
```

Comment:
It seems that this function does nothing:

1. It has no `else` branch that would fail if `type` is `nil`.
2. `<schema node>.fields` is passed to it, not a schema node object.

Reply:
> It seems that this function does nothing:

This function checks that all fields of `cluster_config.schema` has a description field.
```lua
local field_path = table.concat(ctx.path, '.')
t.assert(schema.description ~= nil, string.format("%q is missing description", field_path))
```


> 1. It has no `else` branch that would fail if `type` is `nil`.

I think we don't expect a nil type here because we're working with `schema`, which doesn't have an implied `nil` value in its structure. So I think we can just recursively iterate over the fields of `cluster_config.schema` and not worry about the `nil` type.

> 2\. `<schema node>.fields` is passed to it, not a schema node object.

Now `<schema node>` passed, not `<schema node>.fields`

Reply:
I think it is too easy to make this test no-op (as it was when `<schema node>.fields` is passed if I understand it right). Anyway, don't want to be picky about a testing code.

The last remark from me: I would place it into test/config-luatest/cluster_config_schema_test.lua.

Reply:
Moved to `test/config-luatest/cluster_config_schema_test.lua`